### ISSUE TITLE
detekt-api test-fixtures runtime JAR is missing from Maven Central since 2.0.0-alpha.3, breaking detekt-test resolution

### ISSUE THREAD

**Issue by be-hase** [detekt/detekt#9543]
## Expected Behavior

  Depending on `dev.detekt:detekt-test` should allow resolving the test classpath, so that rule tests can be written using the helpers it exposes (e.g. `dev.detekt.test.lint`). `detekt-test` requests the
  `dev.detekt:detekt-api-test-fixtures` capability from `detekt-api`, so `detekt-api` should publish the consumable test-fixtures artifact (`detekt-api-<version>-test-fixtures.jar`) together with `testFixturesApiElements` /
  `testFixturesRuntimeElements` variants in its Gradle Module Metadata.

  ## Observed Behavior

  Resolving `testRuntimeClasspath` fails because `detekt-api` only publishes a **documentation-only** (sources) test-fixtures variant — the consumable test-fixtures library variant and its JAR are missing from Maven Central.

  ```
  > Could not resolve all files for configuration ':...:testRuntimeClasspath'.
     > Could not resolve dev.detekt:detekt-api:2.0.0-alpha.5.
       Required by:
           project ':...' > dev.detekt:detekt-test:2.0.0-alpha.5
        > No matching variant of dev.detekt:detekt-api:2.0.0-alpha.5 was found.
          The consumer was configured to find a library for use during runtime,
          compatible with Java 17, packaged as a jar, ... with attribute
          'org.jetbrains.kotlin.platform.type' with value 'jvm' but:
            - Variant 'apiElements' ...
            - Variant 'runtimeElements' ...
            - Variant 'sourcesElements' declares documentation ...
            - Variant 'testFixturesSourcesElements' declares documentation ...
  ```

  `detekt-test:2.0.0-alpha.5`'s `runtimeElements` requests the test-fixtures capability of `detekt-api`:

  ```json
  {
    "group": "dev.detekt",
    "module": "detekt-api",
    "version": { "requires": "2.0.0-alpha.5" },
    "requestedCapabilities": [
      { "group": "dev.detekt", "name": "detekt-api-test-fixtures" }
    ]
  }
  ```

  But `detekt-api:2.0.0-alpha.5`'s `.module` metadata only contains a documentation (sources) test-fixtures variant — there is no `testFixturesApiElements` / `testFixturesRuntimeElements`:

  ```
  apiElements                  | (implicit capability)
  runtimeElements              | (implicit capability)
  sourcesElements              | (implicit capability)
  testFixturesSourcesElements  | caps: [dev.detekt:detekt-api-test-fixtures:2.0.0-alpha.5]
  ```

  Confirmed on Maven Central — the runtime test-fixtures JAR is 404 while only the sources JAR exists:

  ```
  $ curl -s -o /dev/null -w "%{http_code}\n" \
      https://repo1.maven.org/maven2/dev/detekt/detekt-api/2.0.0-alpha.5/detekt-api-2.0.0-alpha.5-test-fixtures.jar
  404

  $ curl -s -o /dev/null -w "%{http_code}\n" \
      https://repo1.maven.org/maven2/dev/detekt/detekt-api/2.0.0-alpha.5/detekt-api-2.0.0-alpha.5-test-fixtures-sources.jar
  200
  ```

  The classes are genuinely needed at runtime — `detekt-test`'s bytecode references `dev.detekt.api.testfixtures.TestDetektionKt`, and the sources JAR contains `TestDetektion`, `TestFactory`, `TestSetupContext` — so this cannot be
  worked around by simply excluding the capability. The same sources-only pattern is present for `2.0.0-alpha.3` and `2.0.0-alpha.4` on Maven Central, i.e. the runtime test-fixtures JAR of `detekt-api` has been missing since `alpha.3`.

  ## Steps to Reproduce

  ```kotlin
  // build.gradle.kts
  dependencies {
      testImplementation("dev.detekt:detekt-test:2.0.0-alpha.5")
  }
  ```

  ```
  ./gradlew test
  ```

  Resolution of the test classpath fails with the "No matching variant" error above.

  ## Context

  We maintain a set of custom detekt rules and use `detekt-test` to unit-test them. When bumping detekt from `2.0.0-alpha.2` to `2.0.0-alpha.5`, the test task can no longer resolve its classpath, so all rule tests are blocked. Because
  the required test-fixtures classes are not published in any consumable form, there is no clean workaround on the consumer side — we're stuck on `alpha.2` until this is fixed.

  ## Your Environment

  * Version of detekt used: `2.0.0-alpha.5` (also reproduced with `2.0.0-alpha.3` and `2.0.0-alpha.4`)
  * Version of Gradle used (if applicable): `9.4.1` and `9.6.1`
  * Gradle scan link (add `--scan` option when running the gradle task): N/A
  * Operating System and version: macOS (Darwin 24.2.0); also reproduced on Linux CI (Ubuntu)
  * Link to your project (if it's a public repository): N/A (private repository)

**Comment by cortinico** [comment]
Thanks for reporting it. Yes this regressed since we moved to vannikteck publishing plugin

Fix is here:
- https://github.com/detekt/detekt/pull/9545

**Comment by be-hase** [comment]
thanks :)
